### PR TITLE
fix(stackdriver): Use metric labels

### PR DESCRIPTION
**Pull request by emzeq**
Adding the metric labels as tags enables comparison of
individual time series when using "Group By"; it currently
only compares the last time series.

### CODE REVIEW

File: kayenta-stackdriver/src/main/java/com/netflix/kayenta/stackdriver/metrics/StackdriverMetricsService.java [hunk 1]
```diff
@@ -414,20 +414,31 @@ public String buildQuery(
...
+      Metric metric = timeSeries.getMetric();
+      if (metric != null) {
+        Map<String, String> labels = metric.getLabels();
+
+        if (labels != null) {
+          filteredLabels.putAll(labels);
         }
       }
 
+      metricSetBuilder.tags(filteredLabels);
```
Comment (fieldju) [inline kayenta-stackdriver/src/main/java/com/netflix/kayenta/stackdriver/metrics/StackdriverMetricsService.java:440]: Whats the difference between what labels are returned by `monitoredResource.getLabels();` vs `metric.getLabels();`?

Comment (emzeq) [reply kayenta-stackdriver/src/main/java/com/netflix/kayenta/stackdriver/metrics/StackdriverMetricsService.java:440]: @fieldju `monitoredResource.getLabels();` returns the labels for the reporting GCP resource (e.g., `project_id`, `pod_id`, `namespace_id`) and are usually automatically added to custom metrics. `metric.getLabels();` returns the labels defined by the metric descriptor and are user-defined for custom metrics (e.g., `status_code`, `controller`, `action`).

**Comment by fieldju** [comment]
Sorry I have been on paternity leave. I am back now. Trying to remember why I didn't approve this when I last looked at code.

File: kayenta-stackdriver/src/main/java/com/netflix/kayenta/stackdriver/metrics/StackdriverMetricsService.java (hunk 1 above)
Comment (fieldju) [reply kayenta-stackdriver/src/main/java/com/netflix/kayenta/stackdriver/metrics/StackdriverMetricsService.java:440]: Is there a reason to not have those be separate things in the metadata?

```metricSetBuilder.attribute("metric-labels, metric.getLabels());```

Comment (emzeq) [reply kayenta-stackdriver/src/main/java/com/netflix/kayenta/stackdriver/metrics/StackdriverMetricsService.java:440]: I didn't find anything in the code that was using attributes for creating comparison pairs.

**Comment by emzeq** [comment]
> Sorry I have been on paternity leave. I am back now. Trying to remember why I didn't approve this when I last looked at code.

Not a problem. Congrats on your new baby!

**Pull request merged**